### PR TITLE
integration tests

### PR DESCRIPTION
**Pull request by benbrandt**
Moves the `test_examples.sh` and `lint_examples.sh` to Rust integration tests.

This is indeed much nicer and makes it possible to actually test the Http and Tcp ones, for which I introduced a bug in my last changes that was now caught 🤦🏻

### CODE REVIEW

File: bundled/poll_loop.py [hunk 1]
```diff
@@ -130,18 +146,18 @@ def run_until_complete(self, future):
...
                 new_wakers = []
                 ready = [False] * len(pollables)
                 for index in poll.poll(pollables):
                     ready[index] = True
-                
+
                 for (ready, pollable), waker in zip(zip(ready, pollables), wakers):
                     if ready:
-                        pollable.__exit__()
+                        pollable.__exit__(None, None, None)
```
Comment (benbrandt) [inline bundled/poll_loop.py:160]: This was the issue I introduced by doing the mypy types to match the usual __exit__ pattern... maybe this was a mistake.

**Comment by benbrandt** [comment]
Ok classic "works on my machine" but not CI... I'll try and dig in. I think I may need to rework some of this

File: tests/componentize.rs [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        .current_dir(dir.path())
+        .args(["-m", "venv", ".venv"])
+        .assert()
+        .success();
+
+    Command::new("./.venv/bin/pip")
+        .current_dir(dir.path())
+        .args(["install", "wasmtime"])
+        .assert()
+        .success();
```
Comment (benbrandt) [inline tests/componentize.rs:213]: @dicej I'm not sure if you have a preference here or not. This makes sure the tests can run as long as you have Python installed. But we could also have some docs in the contributing section that python + wasmtime should already be setup. We'll have the same issue with mypy for linting.

**Comment by benbrandt** [comment]
@dicej ok after much wrestling with windows, this should be ready. All example running and linting is now happening in Rust tests, and with that I was able to get much better coverage.

File: tests/componentize.rs [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+    Ok(())
+}
+
+fn venv_path(path: &Path) -> PathBuf {
+    path.join(".venv")
+        .join(if cfg!(windows) { "Scripts" } else { "bin" })
+}
+
+fn install_numpy(path: &Path) {
```
Comment (dicej) [inline tests/componentize.rs:295]: It's slightly annoying we have two copies of this function and `venv_path` across `bindings.rs` and `componentize.rs`, but I'm not sure it's worth consolidating them into their own library crate.  Something to revisit if the amount of duplication increases over time.

**Review by dicej — APPROVED** [review]
Looks great overall.  Thanks so much for doing this!

As I mentioned inline, I tend to prefer we use library crates and/or `std` over shelling out to `curl`, `tar`, and `nc` since it reduces the number of utilities that need to be installed to run the tests.  Another advantage is that we would have more precise control for the HTTP and TCP examples: rather than telling e.g. `curl` to do timed retries while waiting for `wasmtime` to start listening for connections, we could programmatically start the server on an emphemeral port and _then_ use `reqwest` to send a request to that port (which also avoids potential port conflicts if we ever have more than one HTTP test running in parallel).  The downside is that there are more steps to setting up an HTTP server and hooking it up to wasmtime vs. just running `wasmtime serve`.  Anyway, none of that is a blocker, but something to keep in mind as we add more examples and/or tests.

I do think it's reasonable to shell out to `pip` and `python` given that it would presumably be more complicated to replace those with library crates.

**Comment by benbrandt** [comment]
Good points. I was so focused on just getting the readme examples to work, I neglected this (at some point I thought of pulling in wasmtime as a lib, but pressed on).

I can do a round two when I get a chance to clean this up and be a little more maintainable. It would also have the benefit of running the `nc` example on Windows.

**Pull request merged**